pkg/healthz/healthz.go

```diff
@@ -17,25 +17,100 @@ limitations under the License.
 package healthz
 
 import (
+	"bytes"
+	"fmt"
 	"net/http"
+	"sync"
 )
 
+var (
+	// guards names and checks
+	lock = sync.RWMutex{}
+	// used to ensure checks are performed in the order added
+	names  = []string{}
+	checks = map[string]*healthzCheck{}
+)
+
+func init() {
+	http.HandleFunc("/healthz", handleRootHealthz)
+	// add ping health check by default
+	AddHealthzFunc("ping", func(_ *http.Request) error {
+		return nil
+	})
+}
+
+// AddHealthzFunc adds a health check under the url /healhz/{name}
+func AddHealthzFunc(name string, check func(r *http.Request) error) {
+	lock.Lock()
+	defer lock.Unlock()
+	if _, found := checks[name]; !found {
+		names = append(names, name)
+	}
+	checks[name] = &healthzCheck{name, check}
+}
+
+// InstallHandler registers a handler for health checking on the path "/healthz" to mux.
+func InstallHandler(mux mux) {
```

Comment:
What about combining these two functions? It seems a bit harder to use for the user to have to know to do:
- Add
- Add
- Install

They may potentially forget the Install (as I did from first reading :) )